ast.d

```diff
@@ -148,6 +150,30 @@ OperatorType parseOperatorType(string input)
     }
 }
 
+enum OperatorClass
+{
+    Math,
+    Relational,
+    Logical
+}
```

Comment:
Could probably make the generate functions for both types render the same assembly but relational operators can get away with no jumps, while logical need jumps to allow stopping evaluation early when the answer is known from the left side only.